### ISSUE TITLE
There should be a way to create a LogFile instance from a full file path

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2575]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2575|
|Type|enhancement|
|Created|2007-04-16 12:04:37Z|

This removes the need of functions like twisted.application.app.getLogfile.

<details><summary>Searchable metadata</summary>

```
trac-id__2575 2575
type__enhancement enhancement
reporter__therve therve
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1176725077000000 1176725077000000
changetime__1177587314000000 1177587314000000
version__None None
owner__ 
cc__exarkun cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve removed owner|
|-|-|

Ready to review in logfile-full-path-2575.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @therve|
|-|-|

Looks good.  Should the code in Twisted which calls app.getLogFile be updated to instantiate LogFile directly with the new factory function instead, and should app.getLogFile get a (pending) deprecation warning?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

I changed the code that uses app.getLogFile, and added a warning.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @dreid<br>@dreid set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid set owner to @therve<br>@dreid set status to `new`|
|-|-|

This looks really good, but I think it'd be useful to be able to do LogFile.fromFullPath(filename, defaultMode=...)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @dreid|
|-|-|

OK, I ended up adding all arguments.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @therve|
|-|-|

blarg, conflicts due to the [#2574](https://github.com/twisted/twisted/issues/2574) merge, can you merge forward and resolve the conflicts?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve removed owner|
|-|-|

OK, merged in logfile-full-path-2575-2.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_10"></a>|@ralphm set owner to @therve|
|-|-|

The last change (arguments) looks good to me, as do the other changes that the other reviewers looked at.

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set status to `closed`|
|-|-|

(In [20074]) Merge logfile-full-path-2575-2

Author: therve
Reviewers: ralphm, exarkun, dreid
Fixes [#2575](https://github.com/twisted/twisted/issues/2575)

Add a fromFullPath class method on LogFile that allows to create a LogFile
instance using the full path to a file. That replaces app.getLogFile, which
becomes deprecated.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
